sunpy/map/mapbase.py

```diff
@@ -460,6 +460,11 @@ def date(self):
             time = self._default_time
         return parse_time(time)
 
+    @date.setter
+    def date(self, val):
+        val = parse_time(val)
+        self.meta['date-obs'] = val.isot
```

Comment:
Is there any extra check we need to do? Do we keep a record somewhere that the date has been changed? What's the motivation for changing the date?

Reply:
I think as long as we're updating the meta we *should* be fine, as that is where all properties draw the map metadata from.